sbot/utils.py

```diff
@@ -199,3 +201,29 @@ def get_USB_identity(port: ListPortInfo) -> BoardIdentity:
         logger.warning(
             f"Failed to pull identifying information from serial device {port.device}")
         return BoardIdentity()
+
+
+def ensure_atexit_on_term() -> None:
+    """
+    Ensure `atexit` triggers on `SIGTERM`.
+
+    > The functions registered via [`atexit`] are not called when the program is
+      killed by a signal not handled by Python
+    """
+
+    if signal.getsignal(signal.SIGTERM) != signal.SIG_DFL:
+        # If a signal handler is already present for SIGTERM,
+        # this is sufficient for `atexit` to trigger, so do nothing.
+        return
+
+    def handle_signal(handled_signum: int, frame: Optional[FrameType]) -> None:
```

Comment:
We're standardising on new-style type annotations.
```suggestion
    def handle_signal(handled_signum: int, frame: FrameType | None) -> None:
```

Reply:
Interesting - [The docs](https://docs.python.org/3/library/typing.html#typing.Optional) claim that's 3.10+